python/ray/serve/http_proxy.py

```diff
@@ -173,10 +175,13 @@ async def __init__(
                                               **middleware.options)
 
         # Start running the HTTP server on the event loop.
-        asyncio.get_event_loop().create_task(self.run())
+        self.running_task = asyncio.get_event_loop().create_task(self.run())
 
-    def ready(self):
-        return True
+    async def ready(self):
+        done, pending = await asyncio.wait(
+            [self.setup_complete.wait(), self.running_task],
+            return_when=asyncio.FIRST_COMPLETED)
+        return await done.pop()
```

Comment:
piggybacked this change while I debug some test failures. Now `serve.start` will no longer hanging for a long time.

Reply:
What was the issue? I don't fully understand.